sars2seq/translate.py

```diff
@@ -133,3 +147,27 @@ def translateSpike(seq):
             current += (subsequentGaps + index)
 
     return sequence
+
+
+def checkSpikeInsertions(accession, seq):
+    """
+    Check and parse out known insertions in the Spike protein.
+
+    @param seq: A C{str} amino acid sequence.
+    """
+    seqLen = len(seq)
+    if seqLen == 1274:
+        # There are no insertions
+        return seq
+    if seqLen > 1274:
+        if 'QTKGIALSPR' in seq:
+            return seq[:679] + seq[683:]
+        elif 'NLVRTDRDLPQ' in seq:
+            return seq[:214] + seq[217:]
+        elif 'LVRAAGYLPQ' in seq:
+            return seq[:214] + seq[217:]
+        else:
+            print(f'Sequence with accession {accession} is too long and '
+                  f'does not have a known insertion.', file=sys.stderr)
+    print(f'Sequence with accession {accession} is too short.',
+          file=sys.stderr)
```

Comment:
It would be better if this raised something too, I think. That gives the caller a way to catch this and also a way to not have the code unconditionally print to stderr.

Reply:
OK, I made it raise an exception.